xbmc/addons/gui/GUIDialogAddonSettings.cpp

```diff
@@ -128,7 +132,134 @@ bool CGUIDialogAddonSettings::ShowForAddon(const ADDON::AddonPtr& addon,
   if (!g_passwordManager.CheckMenuLock(WINDOW_ADDON_BROWSER))
     return false;
 
-  if (!addon->HasSettings())
+  bool byNew = false;
+  uint32_t instanceNumber = KODI::ADDONS::ADDON_SETTINGS_ID;
+
+  /*
+   * Below becomes the optional possibility handled within an selection dialog to
+   * select about one add-on given different instances, e.g. by PVR addons to
+   * allow various connections to backends.
+   */
+  if (addon->SupportInstanceSettings())
+  {
+    constexpr int BASE_SETTINGS = 0;
+    constexpr int ADD_INSTANCE = 100;
+    constexpr int DELETE_INSTANCE = 101;
+    constexpr int GENERAL_BUTTON_START = ADD_INSTANCE;
+
+    CGUIDialogSelect* dialog =
+        CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+            WINDOW_DIALOG_SELECT);
+    if (dialog)
+    {
+      while (true)
+      {
+        CFileItemList itemsGeneral;
+        CFileItemList itemsInstances;
+
+        dialog->Reset();
+        dialog->SetHeading(10012); // Label: "Add-on configuration setting selection"
+        dialog->SetUseDetails(false);
+
+        if (addon->HasSettings(KODI::ADDONS::ADDON_SETTINGS_ID))
+        {
+          // Label: "Base settings"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10013));
+          item->SetProperty("id", BASE_SETTINGS);
+          itemsGeneral.Add(item);
+        }
+        {
+          // Label: "Add add-on configuration"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10014));
+          item->SetProperty("id", ADD_INSTANCE);
+          itemsGeneral.Add(item);
+        }
+        {
+          // Label: "Delete add-on configuration"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10015));
+          item->SetProperty("id", DELETE_INSTANCE);
+          itemsGeneral.Add(item);
+        }
+
+        std::vector<uint32_t> ids = addon->GetKnownInstanceIds();
+        uint32_t highestId = 0;
+        for (const auto& id : ids)
+        {
+          // Format label: "Add-on configuration #{0:d}"
+          const std::string label = StringUtils::Format(g_localizeStrings.Get(10017), id);
```

Comment:
We should display the configuration's title here, not the id, which says nothing to the user